lib/vsc/zk/rsync/source.py

```diff
@@ -312,10 +314,15 @@ def get_flags(self, file, recursive):
             flags.append('--delete')
         if self.rsync_checksum:
             flags.append('--checksum')
+        if self.rsync_hardlinks:
+            flags.append('--hard-links')
         if self.rsync_verbose:
             flags.append('--verbose')
         if self.rsync_dry:
             flags.append('-n')
+        if self.rsync_arbitopts:
+            for arbopt in self.rsync_arbitopts:
```

Comment:
make a temporary array with the flags, log all the arbitrary options with a `log.warning` and then do `flags.extend(arb_flags)`